src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,203 @@
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsIdsByDriverQuery = "SELECT * from cars_drivers WHERE driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(getCarsIdsByDriverQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(get(resultSet.getObject("car_id", Long.class)).orElseThrow(
+                        () -> new RuntimeException("Couldn't get cars for driver with id "
+                                + driverId)));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars for driver with id "
+                    + driverId, e);
+        }
+        return cars;
+    }
+
+    private static void deleteConnectionsCarsDrivers(Car car) {
+        String deleteConnectionsQuery = "DELETE FROM taxi_service_db.cars_drivers "
+                + "WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteConnectionsStatement
+                        = connection.prepareStatement(deleteConnectionsQuery)) {
+            deleteConnectionsStatement.setLong(1, car.getId());
+            deleteConnectionsStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+    }
+
+    private void insertDrivers(Car car) {
+        String insertDriversQuery = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(insertDriversQuery)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver: car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't insert driver into car: "
+                    + car + ". ", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("car_id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String name = resultSet.getString("name");
+        String country = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerId, name, country);
+        return new Car(id, model, manufacturer, null);
```

Comment:
passing nulls into constructor is a bad practice
create constructor with smaller number of parameters or use setters